avocado/utils/podman.py

```diff
@@ -41,6 +56,119 @@ def __init__(self, podman_bin=None):
 
         self.podman_bin = path
 
+
+class Podman(_Podman):
+    def execute(self, *args):
+        """Execute a command and return the returncode, stdout and stderr.
+
+        :param args: Variable length argument list to be used as argument
+                      during execution.
+        :rtype: tuple with returncode, stdout and stderr.
+        """
+        try:
+            LOG.debug("Executing %s", args)
+
+            cmd = [self.podman_bin, *args]
+            proc = subprocess.Popen(
+                cmd,
+                stdin=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            stdout, stderr = proc.communicate()
+            LOG.debug("Return code: %s", proc.returncode)
+            LOG.debug("Stdout: %s", stdout.decode("utf-8", "replace"))
+            LOG.debug("Stderr: %s", stderr.decode("utf-8", "replace"))
+        except (FileNotFoundError, PermissionError) as ex:
+            # Since this method is also used by other methods, let's
+            # log here as well.
+            msg = "Could not execute the command."
+            LOG.error("%s: %s", msg, str(ex))
+            raise PodmanException(msg) from ex
+
```

Comment:
In the [async version,](https://github.com/avocado-framework/avocado/pull/5870/commits/bfccea3f250d1f12074078f5edb0c2c075db0846#diff-54aaa23aa947160398e8f53946a877faccbdfa6dbbfa40adf80b7a61a01457b5R199) we raise `PodmanException` if the return code is not 0. IMO we should have the same behaviour on sync version as well.

Reply:
Good catch @richtja , thanks for spotting it.